presto-iceberg/src/test/java/io/prestosql/plugin/iceberg/TestIcebergSmoke.java

```diff
@@ -387,177 +387,203 @@ private void testCreatePartitionedTableAs(Session session, FileFormat fileFormat
 
 //        assertEquals(partitions.size(), 3);
 
-        assertQuery(session, "SELECT * from test_create_partitioned_table_as", "SELECT orderkey, shippriority, orderstatus FROM orders");
+        assertQuery("SELECT * from test_create_partitioned_table_as", "SELECT orderkey, shippriority, orderstatus FROM orders");
 
-        dropTable(session, "test_create_partitioned_table_as");
+        dropTable("test_create_partitioned_table_as");
```

Comment:
`dropTable` calls should be called always at the end of the tests. Currently if assertion fails the `DROP` will not be executed and test for another file format which uses same table name will fail on `CREATE TABLE`

Reply:
Thanks for taking a look, @losipiuk!  AFAIK `dropTable` is always called, except as you point out in the case of an exception.  That was also true before these changes?

Are you suggesting that all tests should be wrapped in `try/finally` with `dropTable` called in the `finally` block?

Reply:
Before changes single table was only used for single test run, right? Then not having drop is not painful. We just end up with garbage. 
Now that we have same test executed multiple times (for different file formats), failure for test in format A will result in failure at the very beginning of test for format B.

I think `try/finally` is fine approach. Though extra indentation makes code a bit less readable.

Or you can use temporary table mechanism which @kokosing  intruduces in https://github.com/prestosql/presto/pull/4849, as soon as that one gets merged.

Reply:
I'll think about it, and consult with @electrum.  If we need to try/finally, I can hide it in a helper method that takes a runnable, e.g., `withCreatedTable(tableName, runnable)`.